src/DataProtection/DataProtection/src/MetricsLoggerProvider.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Metrics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Xml.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Win32;
+
+namespace Microsoft.AspNetCore.DataProtection;
+
+internal sealed class MetricsLoggerProvider : ILoggerProvider
+{
+    public const string MeterName = "Microsoft.AspNetCore.DataProtection";
+
+    private readonly Meter? _meter;
+
+    private readonly ILogger _logger;
+
+    public MetricsLoggerProvider()
+    {
+        _logger = NullLogger.Instance;
+    }
+
+    public MetricsLoggerProvider(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.Create(MeterName);
+
+        var errorCounter = _meter.CreateCounter<long>(
+            "errors",
```

Comment:
I imagine there are standardized names and units for these.

Reply:
The general guidance for Meter and instrument names: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation#best-practices

and for units: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation#best-practices-3